src/js/libcs/Operators.js

```diff
@@ -4320,22 +4320,37 @@ evaluator.format$2 = function(args, modifs) { //TODO Angles
     var v1 = evaluateAndVal(args[1]);
     var dec;
 
-    function fmtNumber(n) {
+    // check if we want to truncate - do so by default
+    var truncate = true;
+    if (modifs.truncate) {
+        var modif = evaluate(modifs.truncate);
+        if (modif.ctype === "boolean")
+            truncate = modif.value;
+    }
+
+    function fmtNumber(n, trunc) {
         var erg = n.toFixed(dec),
             erg1;
+
         do {
             erg1 = erg;
             erg = erg.substring(0, erg.length - 1);
-        } while (erg !== "" && erg !== "-" && +erg === +erg1);
-        return "" + erg1;
+        } while (trunc && erg !== "" && erg !== "-" && +erg === +erg1);
+
+        var tmp = "" + erg1;
+        // switch delimiter if needed
+        if (modifs.delimiter && modifs.delimiter.ctype === "string") {
+            tmp = tmp.replace(".", modifs.delimiter.value);
+        }
+        return tmp;
     }
 
     function fmt(v) {
         var r, i, erg;
         if (v.ctype === 'number') {
-            r = fmtNumber(v.value.real);
-            i = fmtNumber(v.value.imag);
-            if (i === "0")
+            r = fmtNumber(v.value.real, truncate);
+            i = fmtNumber(v.value.imag, truncate);
+            if (v.value.imag === 0.0)
```

Comment:
What about
```suggestion
            if (i === fmtNumber(0, truncate))
```
maybe also the case ` i===-0` should be checked as well.